Apps/Immunization/src/Services/VaccineStatusService.cs

```diff
@@ -131,5 +140,92 @@ public async Task<RequestResult<VaccineStatus>> GetVaccineStatus(string phn, str
 
             return retVal;
         }
+
+        /// <inheritdoc/>
+        public async Task<RequestResult<ReportModel>> GetVaccineStatusPDF(string phn, string dateOfBirth)
+        {
+            RequestResult<VaccineStatus> vaccineStatusResult = await this.GetVaccineStatus(phn, dateOfBirth).ConfigureAwait(true);
+
+            if (vaccineStatusResult.ResultStatus == ResultType.Success)
```

Comment:
You may also want to check that the ResourcePayload?.Result.StatusIndicator is not empty.